added script and emphasis to audio sample payload

Waidhoferj opened the pull request:
I added an *emphasis* and *script* field to the sql call of the REST API. I also took out a validation check in `save_audio_sample_meta_data()` that would reject data with too much information. Since we would only take a subset of `formatted_data`, then we should only reject the data if there is too little information.

Commit:
added script and emphasis to audio sample payload

snekiam reviewed (approved):
This looks good to me.  I didn't see the AudioSampleMetadata table in the database (so I don't know if the columns are correct), but everything else looks good!

mfekadu commented:
@Waidhoferj  would you please provide a **quick screenshot** that shows the code working as you expect? Then, please feel free to click the **merge pull request** button yourself.

mfekadu commented on Entity/AudioSampleMetaData.py:
```diff
@@ -26,6 +26,8 @@ class AudioSampleMetaData(Base):
     tone = Column(String(255))
     timestamp = Column(Integer)
     username = Column(String(255))
+    script = Column(String(1024))
```
I am surprised that `1024` is allowable in MySQL

also, what is a "script"  in the context of the `AudioSampleMetaData`?

Waidhoferj replied:
The script is something I added with the latest version of the CSAI Recorder. Since we now know what people are saying given the prompt, we can store the script and use this training set in other NLP projects

Waidhoferj replied:
I'm not yet set up with the database. Lets get that working tomorrow

mfekadu reviewed (commented):
briefly reading through the code, looks good. 

just need a quick test case or screenshot of the database

mfekadu commented:
# Altered the database `dev_feb_3_2020` to include `script` and `emphasis` fields

<img width="1134" alt="image" src="https://user-images.githubusercontent.com/11131250/75117242-92cde580-5624-11ea-8fd4-5bc88219ba35.png">

mfekadu commented:
# tested on `dev_feb_3_2020` database and `save_audio_sample_meta_data` seems to INSERT properly

![Feb-23-2020 10-11-12](https://user-images.githubusercontent.com/11131250/75117279-e50f0680-5624-11ea-860a-f1c5a9e4ba10.gif)

mfekadu commented:
# Sorry I looked more closely and realized that `emphasis` was not saved properly... fixed with 1 line of code
<img width="1142" alt="image" src="https://user-images.githubusercontent.com/11131250/75117345-52229c00-5625-11ea-86de-d98a08272222.png">

```
metadata.emphasis = formatted_data['emphasis']
```

mfekadu committed:
make sure emphais is INSERTED && document function better

manually tested

(venv) ➜  api git:(csai-recorder-script-added) ✗ ./database_wrapper.py
initialized database session
initialized NimbusMySQLAlchemy
course_list: [<Courses (dept=CSC, course_num=357)>]
course_list: [<Courses (dept=CSC, course_num=357)>]
course_list[0].courseName CSC 357. Systems Programming.
table already exists
{'isWakeWord': 'ww', 'firstName': 'john', 'lastName': 'doe', 'gender': 'f', 'noiseLevel': 'q', 'location': 'here', 'tone': 'serious-but-not-really', 'timestamp': 1577077883, 'username': 'guest', 'filename': 'filename', 'script': '??????????', 'emphasis': 'iss'}
NimbusMySQLAlchemy closed
(venv) ➜  api git:(csai-recorder-script-added) ✗

mfekadu commented on database_wrapper.py:
```diff
@@ -338,6 +338,8 @@ def save_audio_sample_meta_data(self, formatted_data: dict) -> bool:
             "tone": "serious-but-not-really",
             "timestamp": 1577077883,
             "username": "guest",
+            "script": "??????????",  # FIXME: what is script?
```
can we include an example of `script`?

mfekadu commented on database_wrapper.py:
```diff
@@ -725,7 +720,9 @@ def close(self) -> None:
         "username": "guest",
-        "filename": "filename"
+        "filename": "filename",
+        "script": "??????????",  # FIXME: what is script?
```
can we include a good example of `script`?

mfekadu commented on database_wrapper.py:
```diff
@@ -350,7 +352,7 @@ def save_audio_sample_meta_data(self, formatted_data: dict) -> bool:
         keys_i_care_about = {
             'isWakeWord', 'firstName', 'lastName', 'gender', 'noiseLevel',
-            'location', 'tone', 'timestamp', 'username', 'filename'
+            'location', 'tone', 'timestamp', 'username', 'filename', 'emphasis'
```
is `script` a key we care about, if so, then could it be included here?

snekiam commented:
Talked to @Waidhoferj, closing.
